From 547b0ffaaef953f7bfaf04a9c94c7c7561426e19 Mon Sep 17 00:00:00 2001
From: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
Date: Wed, 3 Jul 2019 10:42:24 +0530
Subject: [PATCH] [REVERT ME] Start MessengerService for all users

Issue: No Notification/Audio for BT SMS

Analysis: MessengerService is now started from
VendorServiceController. Earlier it was started from
BOOT_COMPLETED action. As per new changes, Messenger
app is started from VendorServiceController and
following parameters are set for the start of service:

<item>com.android.car.messenger/.MessengerService#bind=
startForeground,user=foreground,trigger=userUnlocked</item>

As per the above change, MessengerService is started only
for Foreground user and also when the user is unlocked.

But in our case, we are trying to start MessengerService
for "System User" i.e. background user and while starting
service, check for current user fails as it should be started
for foreground users as per XML definition.

Fix: Changing the user to "all" instead of Foreground user.
This will start Messenger service for all users i.e.
foreground and background users.

Change-Id: I179aaeaf4888367b45b285c415f65f0f3f265fd3
Tracked-On: https://jira.devtools.intel.com/browse/OAM-84028
Signed-off-by: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
---
 service/res/values/config.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/service/res/values/config.xml b/service/res/values/config.xml
index dffe9b98..6d68ba93 100644
--- a/service/res/values/config.xml
+++ b/service/res/values/config.xml
@@ -177,7 +177,7 @@
          is no longer foreground.
      -->
     <string-array translatable="false" name="config_earlyStartupServices">
-        <item>com.android.car.messenger/.MessengerService#bind=startForeground,user=foreground,trigger=userUnlocked</item>
+        <item>com.android.car.messenger/.MessengerService#bind=startForeground,user=all,trigger=userUnlocked</item>
     </string-array>
 
     <!-- The consent activity that must be shown for every unknown mobile device before projection
-- 
2.17.1

